perspective_rectification/abrupt_corners.py

```diff
@@ -0,0 +1,164 @@
+    """
+    Select the two middle points from the abrupt changes
+    @param point_start: corner where the first abrupt change is detected
+    @param point_end: corner where the next abrupt change is detected related next to the point_start
+    @param len_points: counter of total amount of corners to calculate the abrupt changes that connects the begging of
+    the list with the end
+    """
+    if point_start > point_end:  # if the abrupt change is before ending the sequence we have to start from the end
+        #  until the firs point in the beginning
+        moved_initial_point = len_points - point_start
+        mid_point = (moved_initial_point + point_end) // 2
+        difference = int((point_end - point_start) * 0.25)
+        marked_point_lower = mid_point - difference - moved_initial_point
+        marked_point_upper = mid_point + difference - moved_initial_point
+        if marked_point_lower < 0:
+            marked_point_lower = len_points + marked_point_lower
+    else:
+        mid_point = (point_start + point_end) // 2
+        difference = int((point_end - point_start) * 0.25)
+        marked_point_lower = mid_point - difference
+        marked_point_upper = mid_point + difference
+    return marked_point_lower, marked_point_upper
+
+
+def get_corners_abrupt_changes(cumulative_difference):
+    """
+    Gather each sequence of corners divided by the abrupt changes and return a list of 4 lists each
+    containing one side
+    """
+    # Sliding window
+    series_smoothed = pd.Series(cumulative_difference)
+
+    time_series_parsed = pd.Series(series_smoothed.values[1:] - series_smoothed.values[:-1],
+                                   index=series_smoothed.index[:-1]).abs()
+
+    threshold_window = .7
+    max_time_series = time_series_parsed.max()
+    print(time_series_parsed > max_time_series * threshold_window)
+
+    rolling_windowed = time_series_parsed.rolling(3, min_periods=1, win_type='parzen').sum()
+    max_rolling_windowed_series = rolling_windowed.max()
+    abrupt_detections = rolling_windowed > max_rolling_windowed_series * threshold_window
+
+    is_new_sequence = True
+    total_cumulative = []
+    for i, element in enumerate(abrupt_detections):
+
+        if not element:
+
+            if is_new_sequence:  # meanwhile there is non-abrupt changes keep gathering points
+                current_cumulative = []
+                counter = 0
+
+                for series_contigous in abrupt_detections[i:]:
+                    if series_contigous:
+                        break
+                    counter += 1
+
+                    current_cumulative.append(i + counter)
+                total_cumulative.append(current_cumulative)
+                is_new_sequence = False
+        else:
+            # Once there is an abrupt change check for the next one
+            is_new_sequence = True
+    # New idea: Sort on the mean cumulative change of the points that meet the threshold
```

Comment:
Could you add this as a TODO instead? Currently it's not clear whether this new idea is already implemented or not. It's fine to have TODO comments in a repository like this, they are essentially already TODO's.

Reply:
oh yea, makes sense